modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ListTypeDefinition/utils.ts

```diff
@@ -7,6 +7,14 @@ export function fixLocaleKeys(name_i18n: LocalizedValue<string>) {
 	const newTranslationsObject: {[key: string]: string} = {};
 
 	for (const [key, value] of Object.entries(name_i18n)) {
+		if (key === 'zh-Hans-CN') {
+			newTranslationsObject['zh_CN'] = value;
+		}
+
+		if (key === 'zh-Hant-TW') {
+			newTranslationsObject['zh_TW'] = value;
+		}
+
```

Comment:
1. You can check if there is the possibility for the user to add languages `zh_HANS` and `zh_HANT` through this [process](https://learn.liferay.com/web/guest/w/dxp/building-applications/core-frameworks/localization/adding-a-language). If so, these new languages are be available to use in `InputLocalized` component?

2. Have you considered exporting the `translationsNormalizer` function for reuse here?

Reply:
Hello @igor-franca,
1. I do not believe this should be left to the customer to add since these are not new languages, but what the backend sends back for zh_CN and zh_TW.
2. I have not considered this. Would you be able to provide steps on how to go about doing this if this is the approach we want to do?
Thank you!

Reply:
> I do not believe this should be left to the customer to add since these are not new languages, but what the backend sends back for zh_CN and zh_TW.

According to this file, the user can add new languages ​​to the portal. And it recommends using [ISO 639](https://simple.wikipedia.org/wiki/Template:ISO_639_name_zh-Hans). 

I was able to test it by adding the new `zh-Hans` language, and as a result I was able to see my new language in the `inputLocalized` component: 

![screenshot-1706208390](https://github.com/liferay-objects/liferay-portal/assets/61854510/66e8bfbb-59dc-4d7b-aae6-1daac534f021)

By continuing with the current solution and the previously merged solution, we are ignoring the fact that the user wants to use zh-Hans as a portal language. I would just like us to keep this in mind when proceeding with this solution, ok? Do you know if this was the subject of any debate?

However, we need to fix the current problem with `zh_TW` and `zh_CN`. Therefore, I am inclined to validate this solution. Since we already have a validated and merged example.

> I have not considered this. Would you be able to provide steps on how to go about doing this if this is the approach we want to do?

It's just export the `translationsNormalizer` function from `inputLocalized`, and add this export in `frontend-js-components-web/.../resources/index.ts`, for exemple:

`export {translationsNormalizer} from './forms/input/InputLocalized'`

And import in `ListTypeDefinition/utils.ts`:

```
import {translationsNormalizer} from "frontend-js-components-web";

export function fixLocaleKeys(name_i18n: LocalizedValue<string>) {
	const newTranslationsObject: {[key: string]: string} = {};

	for (const [key, value] of Object.entries(name_i18n)) {
		newTranslationsObject[key.replace(/-/g, '_')] = value;
	}

	return translationsNormalizer(newTranslationsObject);
}
```

What do you think of this change?

Reply:
> I am not aware of any debate of adding zh-Hans. It is my understanding that the customer is looking to resolve the issue in https://liferay.atlassian.net/browse/LPD-15379.

Hey @fortunatomaldonado, I'm familiar with this ticket and the reported issue, but I'm trying to think broadly and assess the impacts of our changes.

> As for the code, I ran into an issue in return translationsNormalizer(newTranslationsObject); where the argument types are off. Argument type {[p: string]: string} is not assignable to parameter type InputLocalizedProp
> Let me know if you believe this there is an easy fix for this as I'm not sure there is.

We can update some types to fix this. 

1. The `fixLocalKeys` function does not receive properties of type `LocalizedValue<string>`. It's wrong, because here we are getting the default `BCP47Language`, using `-` instead of `_`.

2. We can check the feasibility to increment the frontend-infra locale type adding `zh_Hans_CN` and `zh_Hant_TW`.

3. At the end, we can type the replace return using the `Liferay.Language.Locale`

I was created an [example](https://github.com/igor-franca/liferay-portal/commit/0467e4b684059d11e5f86cd86e88d92f7ccda4cb) to shared with you, take a look and tell me what you think of these changes.

Reply:
Hello @igor-franca,

Understand completely, thank you for taking time to look into this!

I was able to follow along and test and see that it resolves the current issue.
I pushed the changes.
Truly appreciate the time taken for this!